### DISCUSSION TITLE
Different k-mer combination generate different results

### DISCUSSION THREAD

**Discussion by Niohuruzh** [ablab/spades#1128] [Q&A]
### Description of bug

Hello,
   I try to find a gene whether is complete in my genome assembled by SPADES based on Illumina data. First I try `spades.py -k 21,33,55,77,99 --isolate -t 46 --cov-cutoff auto -1 ${a}_1.rd.fastq -2 ${a}_2.rd.fastq -o output` to generate a genome. Then  I use blastn, and I get a complete gene sequence. Second, I try `spades.py -k 17,33,43,55,77,99,111 --isolate -t 46 --cov-cutoff auto -1 ${a}_1.rd.fastq -2 ${a}_2.rd.fastq -o output` to generate a genome. Then I use blastn, and I get a fragmental gene sequences in different 'NODE'. So is there any suggestion to choose the best k-mer combination when assemble a genome.

Looking forward your reply!
Best wishes

### spades.log

if need

### params.txt

if need

### SPAdes version

v3.15.4

### Operating System

MINT20

### Python Version

Python 3.7.12

### Method of SPAdes installation

conda

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment]
How long are your reads? And what is the average coverage? In general, we suggest to stick to defaults unless one is 100% sure what they are doing and what impact of the assembly is done

**Reply by Niohuruzh** [reply]
135bp; 3G PE data for 25M genome; about 71X depth.
BTW, the defaults means I shouldn't set the k value? I'll try it with `spades.py --isolate -t 40 --cov-cutoff auto -1 ${a}_1.rd.fastq -2 ${a}_2.rd.fastq -o defalut.out`

**Reply by asl** [reply]
Well, using such big k-mer length for such short reads is not recommended. Just use default k-mer lengths auto-selected by assembler

**Reply by Niohuruzh** [reply]
So which options `--isolate` and `--careful` are you recommended to use in this case?

**Reply by asl** [reply]
Start with `--isolate`

**Reply by Niohuruzh** [reply]
Thanks. But why not `--careful`?  Because I have high-coverage data?

**Reply by Niohuruzh** [reply]
OK. It's not SPAdes problem. I use the wrong DNA sequences to `blastn`. Sorry to waste your time. Thanks so much!!!
